python_modules/libraries/dagster-k8s/dagster_k8s/container_context.py

```diff
@@ -96,44 +112,80 @@ def __new__(
             resources=check.opt_mapping_param(resources, "resources"),
             scheduler_name=check.opt_str_param(scheduler_name, "scheduler_name"),
             security_context=check.opt_mapping_param(security_context, "security_context"),
-            server_k8s_config=UserDefinedDagsterK8sConfig(
-                **check.opt_mapping_param(server_k8s_config, "server_k8s_config")
-            ).to_dict(),
-            run_k8s_config=UserDefinedDagsterK8sConfig(
-                **check.opt_mapping_param(run_k8s_config, "run_k8s_config")
-            ).to_dict(),
+            server_k8s_config=server_k8s_config or UserDefinedDagsterK8sConfig.from_dict({}),
+            run_k8s_config=run_k8s_config or UserDefinedDagsterK8sConfig.from_dict({}),
             env=[
                 k8s_snake_case_dict(kubernetes.client.V1EnvVar, e)
                 for e in check.opt_sequence_param(env, "env")
             ],
         )
 
     def _merge_k8s_config(
-        self, first_config: Mapping[str, Any], second_config: Mapping[str, Any]
-    ) -> Mapping[str, Any]:
+        self,
+        onto_config: UserDefinedDagsterK8sConfig,
+        from_config: UserDefinedDagsterK8sConfig,
+    ) -> UserDefinedDagsterK8sConfig:
         # Keys are always the same and initialized in constructor
-        assert set(first_config) == set(second_config)
+        merge_behavior = from_config.merge_behavior
```

Comment:
is this right that only the "from" should control merge behavior? not obvious to me on first glance

Reply:
yeah, "merge_behavior" = "how things should behave when this config is merged into some other config". Can make that clear in the description in the config schema